### DISCUSSION TITLE
QfieldCloud and shared local datasets

### DISCUSSION THREAD

**Discussion by S1lk3** [opengisch/QField#4276] [Q&A]
I use QGIS 3.22.12, QField 2.7.6 on android (tested with both phone and tablet) and the QfieldCloud.

I built a project in QGIS with 2 vector layers (in geopackages), 2 tables (in the same geopackage) and 1 raster file. The raster file is 12GB in size (*.mbtiles based on google satellite and created with "Generate XYZ Tiles" (tile format = JPG)).

Since the raster file is to serve as the basemap for several QField projects that will be used offline, I would like to use it as a "shared local dataset" (https://docs.qfield.org/how-to/outside-layers/?h=basemap). Here, however, I do not succeed.

In detail what I have done:

**In QGIS**
- added 2 vector layers and 2 tables
- added the raster file (*.mbtiles) from folder "C:/Prive/basiskaarten"
- QGIS > Settings > Options > Datasources: localized data paths -> added "C:/Prive/basiskaarten". (printscreen 1)
- Project Properties - QField: "satellite" -> directly access data source + added as basemap (see printscreen 2)
- all layers and the project itself have project system with EPSG: 3857

![image](https://github.com/opengisch/QField/assets/133006424/94daeb3b-3b40-4bdc-a4bf-165fb7ef71c7)
![image](https://github.com/opengisch/QField/assets/133006424/c28d7586-7671-4e69-8926-9c3fc23248da)

**On my android device**: 
- Copied the *.mbtiles file from my desktop to the folder "...\Android\data\ch.opengis.qfield\files\QField\basemaps" with the exact same name.

I save my QGIS project as a *.qgs file (also tried with *.qgz but with a *.qgs I can check if my raster file is effectively saved as localized) and create a new QFieldCloud project ("Convert currently open project to cloud project").

On the qfield cloud webpage, I then see the following error message under uploaded project > jobs: "_invalid_dataprovider Provider is not valid (provider: gdal, URI:_". (printscreen 3) In Qfield itself, I then see the exact same error message when opening the project. The other layers do load.

![image](https://github.com/opengisch/QField/assets/133006424/fe33fe89-f487-489f-8518-4b2ad5dbf36f)

The error message seems incomplete and I have no idea what I am doing wrong. Anyone a suggestion? Thanks in advance.

**Comment by S1lk3** [comment]
Perhaps interesting to add: if I export the project with all files via "package for QField", and import it into QField like this (i.e. without the QField Cloud) then it does manage to load the *.mbtiles file. So I think the problem is not linked to the file itself.

**Comment by S1lk3** [comment]
I now see that it was mentioned here (https://github.com/opengisch/QField/discussions/2736) a year ago that this has yet to be implemented for qfieldcloud. Can an update be given on this please?

Is there a workaround to get my 12GB satellite image into my QfieldCloud project, by manually copying it to a folder on my QField device? It is not desirable to put this heavy file in the cloud given the high cost. I do want the vector files in the cloud; I want to be able to pull in the changes daily via QFieldCloud.

**Comment by dougpiercey** [comment]
Followed the same process and having the same issue, but with a 10GB satellite image .tif. Also looking to avoid storing this in the cloud due to cost and efficiency.

**Comment by dougpiercey** [comment]
Any suggestions on this? Major issue for us.

**Comment by suricactus** [comment]
Support for localized datasets with QFieldCloud is coming in the next weeks as a new feature. Will update you when implemented.

**Reply by dougpiercey** [reply]
Thanks for the update!

**Reply by dougpiercey** [reply]
Hi...just checking in on progress with this. Thanks

**Comment by NellySDIS30** [comment]
Hi, just checking in to know when this functionnality is going to be implemented...

**Comment by max-mapper** [comment]
I tried today on Android to follow the instructions here: https://docs.qfield.org/how-to/outside-layers/. I created a project on QGIS desktop with a `basemap.mbtiles` file that resided in the folder I added into the Localized Data Paths. I confirmed that file was not synced when I synced it to QfieldCloud. Then I manually placed `basemap.mbtiles` into my devices `ch.opengis.qfield/files/QField/basemaps/` folder. But when I open the synced project in QField it my `basemap.mbtiles` does not show up in the layers panel. Interestingly, when I try QField Unstable from the Play Store (which turns out to be a build from December 2022) the `basemap.mbtiles` layer shows up in the layer panel but does not render anything. I am pretty sure this page is inaccurate https://docs.qfield.org/how-to/outside-layers/ and this functionality is totally broken (regardless of if you use QFieldCloud or not) unless someone can point me at a working example

**Reply by meyerlor** [reply]
@maxogden i work with localized data (raster) paths on many projects. It is working perfectly when syncing via cable. Syncing via cloud does not support local datasets so far, that is what is stated here in this thread.
If you create a local data path in QGIS and put your raster in there, you need to remove the raster from the QGIS project, restart QGIS and then add the raster to the project. Save the project, then sync via cable. To verify if QGIS/QFieldSync recognizes the raster data as coming from a localized path, in QFieldSync options it should look like in my screenshot:
![grafik](https://github.com/opengisch/QField/assets/15894401/01391b03-7cec-465e-b8c3-e33526204176)

**Comment by S1lk3** [comment]
Still no update on this issue? Also blocking issue here, qfield cloud is useless to us without this feature, since we work in remote areas.

**Comment by janekska** [comment]
+1

**Comment by dpiercey** [comment]
Hi...any update on the implementation of this? Is it going to happen? If not, we will have to cancel our account.

**Comment by threalslimtobeeee** [comment]
We, as a little NGO working in remote rainforest areas, would also be very happy about this.

**Reply by threalslimtobeeee** [reply]
Another suitable workaround for us would be:
A Project which is flagged to be used as basemap, would also be the basemap of other Projects in Qfield. Not just of standalone datasets. (see #4311 )

**Comment by meyerlor** [comment]
somthing just got merged :)
https://github.com/opengisch/qfieldcloud/pull/817

**Comment by S1lk3** [comment]
Any update on this? I just tested and I still get the same error..

**Reply by aghiraldi** [reply]
Same for me: the error is still there or back (tested last week).

**Reply by jguelat** [reply]
This worked really well between January and mid-March, then something got broken and I cannot get it to work again. I've submitted a ticket yesterday on the QFieldCloud website.

**Reply by suricactus** [reply]
Hey, I am sorry for the regression introduced in late March! Please try again now, we did a small change that should have fixed this issue again.

I would recommend you re-upload your QGIS file (no modifications needed, just slightly move the map canvas in QGIS and reupload the .qgs file) so all the bad state from your project is cleared.

**Reply by jguelat** [reply]
It's working again, thanks!

**Reply by aghiraldi** [reply]
Thanks ! It works great now !

**Reply by suricactus** [reply]
Thanks a lot for confirmations! And again, apologies for the regression. It is great to see people using this functionality in the wild :)

**Reply by meyerlor** [reply]
great stuff! working fine again :)
